src/java/main/org/apache/zookeeper/client/ZooKeeperSaslClient.java

```diff
@@ -215,18 +251,22 @@ public void processResult(int rc, String path, Object ctx, byte data[], Stat sta
         }
     }
 
-    private SaslClient createSaslClient(final String servicePrincipal,
-                                                     final String loginContext) throws LoginException {
+    public void reCreateSaslClient(final String servicePrincipal, final String loginContext) throws LoginException {
```

Comment:
If we do end up agreeing that we need this call, then we should at least make it package protected.